app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,321 @@
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/app/controlplane/plugins/sdk/v1"
+	"github.com/go-kratos/kratos/v2/log"
+)
+
+// Integration implements a generic webhook integration
+type Integration struct {
+	*sdk.FanOutIntegration
+	client *http.Client
+}
+
+// registrationRequest defines the configuration required during registration
+type registrationRequest struct {
+	URL string `json:"url" jsonschema:"minLength=1,description=Webhook URL to send payloads to"`
+}
+
+// attachmentRequest defines the configuration required during attachment
+type attachmentRequest struct {
+	SendAttestation *bool `json:"send_attestation,omitempty" jsonschema:"description=Send attestation (default: true)"`
+	SendSBOM        *bool `json:"send_sbom,omitempty" jsonschema:"description=Send SBOM (default: false)"`
+}
+
+// attachmentState defines the state stored after attachment
+type attachmentState struct {
+	SendAttestation bool `json:"send_attestation"`
+	SendSBOM        bool `json:"send_sbom"`
+}
+
+// registrationState defines the state stored after registration
+type registrationState struct {
+	// No additional state needed for webhook besides the URL stored in credentials
+}
+
+// webhookPayload defines the JSON schema for the webhook payload
+type webhookPayload struct {
+	Metadata *sdk.ChainloopMetadata `json:"Metadata"`
+	Data     json.RawMessage        `json:"Data"` // e.g., SBOM or attestation JSON
+	Kind     string                 `json:"Kind"` // e.g., "SBOM_CYCLONEDX", "ATTESTATION"
+}
+
+// New initializes the webhook integration
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanOut(
+		&sdk.NewParams{
+			ID:          "webhook",
+			Version:     "1.0",
+			Description: "Send Attestation and SBOMs to a generic webhook URL as JSON payloads",
+			Logger:      l,
+			InputSchema: &sdk.InputSchema{
+				Registration: registrationRequest{},
+				Attachment:   attachmentRequest{},
+			},
+		},
+		// Subscribe to SBOMs only, attestations are handled separately
```

Comment:
This comment is not 100% accurate. I'd call it `In Addition to the attestation payload, the following material types will be also present`